Panic: make tfgen: An assertion has failed: strings.HasPrefix(rawname, pkg+"_")

### What happened?

When upgrading to v3.60.0 of the bridge, both https://github.com/pulumi/pulumi-external and https://github.com/pulumi/pulumi-http fail to find the correct resource token.

### Example

Stack trace of update attempt:

```
panic: fatal: An assertion has failed: strings.HasPrefix(rawname, pkg+"_")

goroutine 1 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.81.0/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assertf(0x0?, {0x3072606?, 0xc00021c680?}, {0x0?, 0x11244e5?, 0x2c31fa0?})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.81.0/go/common/util/contract/assert.go:35 +0xe8
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.withoutPackageName({0x302b990?, 0x8}, {0x302b990, 0x8})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/generate.go:1651 +0x7e
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.getMarkdownNames({0x302b990, 0x8}, {0x302b990, 0x8}, 0x0)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/docs.go:189 +0x47
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.getMarkdownDetails({0x3682398, 0xc0009a2840}, {0x302f7f2?, 0xa?}, {0x3043653, 0x13}, {0x302b990, 0x8}, {0x302b990, 0x8}, ...)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/docs.go:234 +0x18c
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.getDocsForResource(0xc000144600, {0x3043653, 0x13}, {0x302b990, 0x8}, {0x302b990, 0x8}, {0x3033659, 0xc}, {0x302b990, ...}, ...)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/docs.go:340 +0x19a
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).gatherDataSource(0xc000144600, {0x302b990, 0x8}, {0x3682fd0?, 0xc0009ec320}, 0xc00097d620)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/generate.go:1402 +0x39b
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).gatherDataSources(0xc000144600)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/generate.go:1349 +0x2f6
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).gatherPackage(0xc000144600)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/generate.go:1025 +0x272
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.(*Generator).Generate(0xc000144600)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/generate.go:850 +0x2a
main.main.Main.func1({{0x302b990, 0x8}, {0x3663560, 0x25}, {0x7ffeefbff95f, 0x6}, {{0x368ff00, 0xc000974ee0}, {0x302b990, 0x8}, ...}, ...})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/main.go:44 +0x65
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.func1(0x0?, {0xc0009a27b0, 0x1, 0x0?})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/main.go:162 +0x658
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.RunFunc.func4(0x0?, {0xc0009a27b0?, 0x0?, 0x0?})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.81.0/go/common/util/cmdutil/exit.go:99 +0x1c
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.newTFGenCmd.RunFunc.RunResultFunc.func6(0xc000a71700?, {0xc0009a27b0, 0x1, 0x3})
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.81.0/go/common/util/cmdutil/exit.go:115 +0x38
github.com/spf13/cobra.(*Command).execute(0xc000005800, {0xc000052090, 0x3, 0x3})
	/Users/guin/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0xc000005800)
	/Users/guin/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/guin/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.MainWithCustomGenerate({_, _}, {_, _}, {{0x368ff00, 0xc000974ee0}, {0x302b990, 0x8}, {0x0, 0x0}, ...}, ...)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/main.go:57 +0x6f
github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen.Main(...)
	/Users/guin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.60.0/pkg/tfgen/main.go:36
main.main()
	/Users/guin/go/src/github.com/pulumi/pulumi-external/provider/cmd/pulumi-tfgen-external/main.go:25 +0xae
make: *** [tfgen] Error 2
```

### Output of `pulumi about`

n/a

### Additional context

pulumi-external is [pinned to a pseudo version of the bridge](https://github.com/pulumi/pulumi-external/blob/main/provider/go.mod#L9) for unknown reasons.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/tfgen/generate.go
+++ b/pkg/tfgen/generate.go
@@ -1652,7 +1652,7 @@ func withoutPackageName(pkg string, rawname string) string {
 	// but every rule has an exception! In HTTP provider the "http" datasource intentionally
 	// does not do that, as noted in:
 	//
-	// https://github.com/hashicorp/terraform-provider-http/blob/0eeb9818e8114631a3c7dc61e750f11180ca987b/internal/provider/data_source_http.go#L47
+	// https://github.com/hashicorp/terraform-provider-http/blob/master/internal/provider/data_source_http.go#L47
 	//
 	// Therefore the code trims the prefix if it finds it, but leaves as-is otherwise.
 	return strings.TrimPrefix(rawname, pkg+"_")

--- a/pkg/tfgen/generate.go
+++ b/pkg/tfgen/generate.go
@@ -1652,7 +1652,8 @@ func withoutPackageName(pkg string, rawname string) string {
 	// but every rule has an exception! In HTTP provider the "http" datasource intentionally
 	// does not do that, as noted in:
 	//
-	// https://github.com/hashicorp/terraform-provider-http/blob/master/internal/provider/data_source_http.go#L47
+	///nolint:lll
+	// https://github.com/hashicorp/terraform-provider-http/blob/0eeb9818e8114631a3c7dc61e750f11180ca987b/internal/provider/data_source_http.go#L47
 	//
 	// Therefore the code trims the prefix if it finds it, but leaves as-is otherwise.
 	return strings.TrimPrefix(rawname, pkg+"_")

--- a/pkg/tfgen/generate.go
+++ b/pkg/tfgen/generate.go
@@ -1648,9 +1648,14 @@ func resourceName(provider string, rawname string,
 
 // withoutPackageName strips off the package prefix from a raw name.
 func withoutPackageName(pkg string, rawname string) string {
-	contract.Assertf(strings.HasPrefix(rawname, pkg+"_"), `strings.HasPrefix(rawname, pkg+"_")`)
-	name := rawname[len(pkg)+1:] // strip off the pkg prefix.
-	return name
+	// Providers almost always have function and resource names prefixed with the package name,
+	// but every rule has an exception! In HTTP provider the "http" datasource intentionally
+	// does not do that, as noted in:
+	//
+	// https://github.com/hashicorp/terraform-provider-http/blob/0eeb9818e8114631a3c7dc61e750f11180ca987b/internal/provider/data_source_http.go#L47
+	//
+	// Therefore the code trims the prefix if it finds it, but leaves as-is otherwise.
+	return strings.TrimPrefix(rawname, pkg+"_")
 }
 
 func stableResources(resources shim.ResourceMap) []string {

--- a/pkg/tfgen/generate_test.go
+++ b/pkg/tfgen/generate_test.go
@@ -365,3 +365,8 @@ func TestModulePlacementForType(t *testing.T) {
 	}
 
 }
+
+func TestWithoutPackageName(t *testing.T) {
+	assert.Equal(t, "http", withoutPackageName("http", "http"))
+	assert.Equal(t, "s3_bucket", withoutPackageName("aws", "aws_s3_bucket"))
+}

